src/metatensor/models/cli/train.py

```diff
@@ -276,18 +305,20 @@ def train_model(
             )
             validation_datasets.append(validation_dataset)
 
-    # Save fully expanded config
+    ###########################
+    # SAVE EXPANDED OPTIONS ###
+    ###########################
+
     OmegaConf.save(
         config=options, f=Path(checkpoint_dir) / "options_restart.yaml", resolve=True
     )
 
     ###########################
-    # SETUP MODEL #############
+    # CREATING DATASET_INFO ###
     ###########################
-    logger.info("Setting up model")
 
-    # TODO: A more direct way to look up the gradients would be to get them from
-    # the configuration dict of the training run.
+    # TODO: A more direct way to look up the gradients would be to get them from the
+    # configuration dict of the training run.
     gradients: Dict[str, List[str]] = {}
     for train_options in train_options_list:
         for key in train_options["targets"].keys():
```

Comment:
Perhaps the extraction of dataset_info from the datasets can be made into its own function

Reply:
I bet we will also need it in eval (when we refine eval)

Reply:
Sure I can think to shift it into a function.

Reply:
I already have a branch for a wrapping this into a `classmethod` of the `DataInfo` class. I will open a request after we merged this PR.